src/extensions/agent_modules/cloud_communication/inc/aduc/mqtt_client.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file mqtt_client.h
+ * @brief Header file for the MQTT client.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+#ifndef __ADUC_MQTT_CLIENT_H__
+#define __ADUC_MQTT_CLIENT_H__
+
+#include "du_agent_sdk/agent_common.h"
+#include "stdbool.h"
+
+EXTERN_C_BEGIN
+
+#define DEFAULT_TCP_PORT 8883
+#define DEFAULT_KEEP_ALIVE_IN_SECONDS 30
+#define DEFAULT_USE_TLS true
+#define DEFAULT_ADPS_CLEAN_SESSION false
+#define DEFAULT_MQTT_BROKER_CLEAN_SESSION true
+#define DEFAULT_MQTT_BROKER_PROTOCOL_VERSION 5
+
+/**
+ * @brief Struct containing the connection settings for the MQTT client.
+ */
+typedef struct ADUC_MQTT_SETTINGS_TAG
+{
+    char* caFile; /*< Path to a PEM file with the chain required to trust the TLS endpoint certificate */
+    char* certFile; /*< Path to a PEM file to establish X509 client authentication */
+    char* clientId; /*< MQTT Client Id */
+    char* hostname; /*< FQDM to the MQTT Broker endpoint, eg: mybroker.mydomain.com */
+    char* keyFile; /*< Path to a KEY file to establish X509 client authentication */
+    char* keyFilePassword; /*< Password (aka pass-phrase) to protect the key file */
```

Comment:
keyFilePassword (and password if cannot be removed) should be fetched JIT and SecureZero from memory right after to protect secrets so should be removed from this struct.

Reply:
Removed.